pkg/agent/client.go

```diff
@@ -64,42 +66,84 @@ func (c *connContext) send(msg []byte) {
 }
 
 type connectionManager struct {
-	mu          sync.RWMutex
+	connLock    sync.RWMutex
 	connections map[int64]*connContext
+
+	// pending dial
+	pendingDial     map[int64]*pendingDialContext
+	pendingDialLock sync.RWMutex
```

Comment:
```suggestion
	pendingDialLock sync.RWMutex
	pendingDial     map[int64]*pendingDialContext
```

nit: move mutex above what it protects